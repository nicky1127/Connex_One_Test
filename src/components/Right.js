import React, { useEffect, useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Paper, Box } from '@material-ui/core';
import client from '../Client';
import Loader from './Loader';

const useStyles = makeStyles(() => ({
  root: {
    width: '100%',
    height: '100%',
    flex: '1 1 auto'
  },
  box: {
    height: '100%',
    overflow: 'scroll'
  }
}));

const Left = () => {
  const classes = useStyles();
  const [loading, SetLoading] = useState(false);
  const [metrics, setMetrics] = useState('');

  const getProm = async () => {
    SetLoading(true);
    try {
      const data = await client._get('/metrics');
      setMetrics(data);
    } catch (err) {
      console.error('Error getting prometheus metrics: ', err);
    }
    SetLoading(false);
  };

  useEffect(() => {
    getProm();
    setInterval(() => getProm(), 30000);
  }, []);

  const renderBox = () => <Box className={classes.box}>{metrics}</Box>;

  const content = loading ? <Loader /> : renderBox();

  return (
    <Paper className={classes.root} elevation={5}>
      {content}
    </Paper>
  );
};

export default Left;
