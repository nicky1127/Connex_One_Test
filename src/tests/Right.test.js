import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Right from '../components/Right';

afterEach(cleanup);

describe('Right', () => {
  it('Snapshot matchs', () => {
    const tree = renderer.create(<Right />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
