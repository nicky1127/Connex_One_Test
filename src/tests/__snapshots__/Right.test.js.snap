// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Right Snapshot matchs 1`] = `
<div
  className="MuiPaper-root makeStyles-root-1 MuiPaper-elevation5 MuiPaper-rounded"
>
  <div
    className="MuiBox-root MuiBox-root-3 makeStyles-box-2"
  >
    
  </div>
</div>
`;
